### DISCUSSION TITLE
[Outdated] [Suggestion] New option "-vv" for interoperation with "--doh-url"

### DISCUSSION THREAD

**Discussion by MasterInQuestion** [curl/curl#12397] [Ideas]
Using ["-v"](https://curl.se/docs/manpage.html#-v) with ["--doh-url"](https://curl.se/docs/manpage.html#--doh-url) gives also the verbose output for DNS requests: which in many cases is unneeded (as in regular "-v" without DoH).
    Using an new option "-vv" to represent the current over-verbose "-v" would probably be desirable, and replacing "-v" to omit DNS details.

**Comment by bagder** [comment]
I think it would more sense (and be closer how other debug/logging/tracing things are already managed) to make a "trace config" out of the DoH outputs so that you could ask for DoH details with `--trace-config DoH`.

**Comment by MasterInQuestion** [comment]
Didn't notice its addition.
    .
    See also:
    https://curl.se/docs/manpage.html#--trace-config
    https://curl.se/libcurl/c/curl_global_trace.html

    Probably..? Do you mean &#x60;--trace-config "-doh"&#x60; alike to accomplish such?
    Though somewhat verbose, but probably applicable.

    Or outright changing DoH debug info into "trace" only?
    Probably also alright.

    Whatsoever the "-vv" looks like a reasonable addition.

    ----

    CLI comparing typical programming languages: ease of input would be emphasized in addition.
    That's why PowerShell style &#x60;`Get-Help Out-String -Full`&#x60; needs shortcut.

**Comment by MasterInQuestion** [comment]
Since 8.7.1 DoH output is default unshown: as without DoH.
    To display the output: use &#x60;`--trace-config "doh"`&#x60; alike.
    .
    [ https://github.com/curl/curl/pull/12411#commits-pushed-44fb5ce](https://github.com/curl/curl/pull/12411#commits-pushed-44fb5ce)

    However then it might be somewhat ambiguous about whether DoH is in effect or not:
    As both on/off may give identical output.

    Some indication would be preferred.
    (maybe just "resolved with DoH" on ["lib/hostip.c"](https://github.com/curl/curl/blob/d0728c9109629ee82b855b350a4c3f1f52ee61df/lib/hostip.c#L1343))

**Reply by bagder** [reply]
A fair suggestion I think!

**Reply by MasterInQuestion** [reply]
In the documentation directly linking to such posts shall also help:
    Which I tend to infinitely maintain and have rigorous quality baseline impose.

**Comment by MasterInQuestion** [comment]
1 question: what's the expectation of "--trace-config" ("curl_global_trace")?
    `curl -v --trace-config "random whatsoever~!@#$%^&*()_+{}:<>?/.,;'[]\=-" -I "https://www.google.com/"`
    ; just worked no complain..?

    Tmp WIP: https://github.com/search?type=code&q=repo:curl/curl+/(?-i)(?:%5CW|^)curl_global_trace(?:%5CW|%24)/

**Reply by icing** [reply]
Yes, intentional. We reserve the possibility to change keywords for tracing in the future. Also, some keywords depend on how curl is built (features). We do not want an invocation to break for having a no-working keyword here.

**Reply by MasterInQuestion** [reply]
I believe there are certain basic syntax that must be followed.
    (I don't think &#x60;curl&#x60; could ever make use of above non-sense alike...)
